pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/documentation/CommentRequiredRule.java

```diff
@@ -109,6 +114,13 @@ private void commentRequiredViolation(Object data, AbstractJavaNode node,
 
     @Override
     public Object visit(ASTClassOrInterfaceDeclaration decl, Object data) {
+        checkCommentMeetsRequirement(data, decl, CLASS_CMT_REQUIREMENT_DESCRIPTOR);
+        return super.visit(decl, data);
+    }
+
+
+    @Override
+    public Object visit(ASTPackageDeclaration decl, Object data) {
         checkCommentMeetsRequirement(data, decl, HEADER_CMT_REQUIREMENT_DESCRIPTOR);
         return super.visit(decl, data);
     }
```

Comment:
This changes now the behavior of "headerCommentRequired": Before it was checking for class javadoc comments, now it checks for package javadoc comments...
This is a incompatible change, that we shouldn't do in a minor release...

Reply:
We won't check package declaration with this rule for the reasons outlined above.